packages/nextjs/src/app-router/server/ClerkProvider.tsx

```diff
@@ -8,14 +10,43 @@ import { ClientClerkProvider } from '../client/ClerkProvider';
 import { initialState } from './auth';
 import { getScriptNonceFromHeader } from './utils';
 
-export function ClerkProvider(props: Without<NextClerkProviderProps, '__unstable_invokeMiddlewareOnAuthStateChange'>) {
+export async function ClerkProvider(
+  props: Without<NextClerkProviderProps, '__unstable_invokeMiddlewareOnAuthStateChange'>,
+) {
   const { children, ...rest } = props;
   const state = initialState()?.__clerk_ssr_state as InitialState;
   const cspHeader = headers().get('Content-Security-Policy');
 
+  const providerProps = { ...mergeNextClerkPropsWithEnv(rest) };
+
+  if (!providerProps.publishableKey) {
+    const ephemeralAccount = await fetchEphemeralAccount();
+    const cookieExpiresAt = cookies().get(constants.QueryParameters.EphemeralExpiresAt)?.value;
+    const cookiePublishableKey = cookies().get(constants.QueryParameters.EphemeralPublishableKey)?.value;
+    const cookieSecretKey = cookies().get(constants.QueryParameters.EphemeralSecretKey)?.value;
```

Comment:
❓ If we assume that we no longer automatically fetch fresh keys, do we need to store the keys in cookies? I think no. We'll have our API return the status of the instance / keys (expired or not) so we can render some sort of callout.

Reply:
I don't think we need to use cookies at all, will do some testing this week.